### PR TITLE
Temporalgrid performance and bugfixes

### PR DESCRIPTION
**Pull request by nerik**
Needs https://github.com/GlobalFishingWatch/mapbox-gl-js/pull/92

- layer-composer: use interaction sourceLayer rather than an extra source 
- layer-composer: heatmap-animated can now use staticTime to avoid spamming MGL worker when zooming out timebar 
- layer-composer: use 'active' time chunk
- react-hooks: useMapInteraction now needs style.metadata as feature.metadata does not seem to be kept up to date by MGL (frame value is wrong) 
- timebar: adds event interaction source in `onChange` callback

### CODE REVIEW

**Review by j8seangel — COMMENTED** [review]
Wow, amazing improvement! It looks way better locally, waiting for the @satellitestudiodesign feedback

>layer-composer: use interaction sourceLayer rather than an extra source 

👏 

> layer-composer: heatmap-animated can now use staticTime to avoid spamming MGL worker when zooming out timebar

 I had some flashes on the date tooltip when moving the bar but it is not important at all of now
Kapture 2020-10-26 at 20 00 19

> layer-composer: use 'active' time chunk

 👏 again, looks clearer now

> react-hooks: useMapInteraction now needs style.metadata as feature.metadata does not seem to be kept up to date by MGL (frame value is wrong) 

nice catch!

> timebar: adds event interaction source in onChange callback 

very useful!

**Comment by j8seangel** [comment]
Ah! I guess you'll have some conflicts with the `selectTimeRange` from [this PR ](https://github.com/GlobalFishingWatch/frontend/pull/172/files#diff-7bbc9ebf0402a6da5c5f5fde0f3c75a379e6d84b178d1d7725925b80adf78a82R5) let me know if you need help to solve them.

File: applications/amathea/package.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
     "@globalfishingwatch/dataviews-client": "^3.3.2",
     "@globalfishingwatch/layer-composer": "^3.0.0",
     "@globalfishingwatch/linting": "^1.1.1",
-    "@globalfishingwatch/mapbox-gl": "1.12.0-gfw.3",
-    "@globalfishingwatch/react-map-gl": "5.2.7-gfw.3",
+    "@globalfishingwatch/mapbox-gl": "1.12.0-gfw.4",
+    "@globalfishingwatch/react-map-gl": "5.2.7-gfw.4",
```
[the remark this answers is not in the document]
Comment (nerik) [reply applications/amathea/package.json:20]: Oh, great!

File: applications/fishing-map/src/features/timebar/timebar.hooks.ts [hunk 2]
```diff
@@ -7,14 +7,27 @@ import {
...
+  const dispatch = useDispatch()
   const { start, end } = useSelector(selectTimerange)
   // TODO needs to be debounced like viewport
-  const dispatchTimerange = (newStart: string, newEnd: string) =>
-    dispatchQueryParams({ start: newStart, end: newEnd })
-  return { start, end, dispatchTimerange }
+  const dispatchTimeranges = (
+    newStart: string,
+    newEnd: string,
+    _: unknown,
```
[the remark this answers is not in the document]
Comment (nerik) [reply applications/fishing-map/src/features/timebar/timebar.hooks.ts:20]: Ok, you're right. I will also send a generic event object to the callback.

File: packages/react-hooks/src/use-map-interaction/use-map-interaction.ts [hunk 3]
```diff
@@ -86,7 +94,8 @@ export const useMapHover = (
   hoverCallbackImmediate?: InteractionEventCallback,
   hoverCallback?: InteractionEventCallback,
   map?: Map,
-  config?: MapHoverConfig
+  config?: MapHoverConfig,
+  metadata?: ExtendedStyleMeta
```
[the remark this answers is not in the document]
Comment (nerik) [reply packages/react-hooks/src/use-map-interaction/use-map-interaction.ts]: Actually, better with an object.

File: applications/amathea/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (nerik) [reply applications/amathea/package.json:20]: @j8seangel See https://github.com/GlobalFishingWatch/frontend/pull/174

**Pull request merged**